### PR TITLE
feat(cce/node_attach): support reset in node_attach

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes NONE

**Special notes for your reviewer**:

**Release note**:

```release-note
support reset in node_attach
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run TestAccCCENodeAttachV3_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccCCENodeAttachV3_basic -timeout 360m -parallel 4
=== RUN   TestAccCCENodeAttachV3_basic
=== PAUSE TestAccCCENodeAttachV3_basic
=== CONT  TestAccCCENodeAttachV3_basic
--- PASS: TestAccCCENodeAttachV3_basic (1303.33s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       1303.427s
```

### CODE REVIEW

File: huaweicloud/resource_huaweicloud_cce_node_attach.go [hunk 1]
```diff
@@ -48,26 +49,22 @@ func ResourceCCENodeAttachV3() *schema.Resource {
...
 				Type:         schema.TypeString,
 				Optional:     true,
-				ForceNew:     true,
 				ExactlyOneOf: []string{"password", "key_pair"},
 			},
 			"password": {
 				Type:         schema.TypeString,
 				Optional:     true,
 				ForceNew:     true,
-				Sensitive:    true,
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_cce_node_attach.go]: why delete the Sensitive attribute?

File: huaweicloud/resource_huaweicloud_cce_node_attach.go [hunk 2]
```diff
@@ -330,7 +388,86 @@ func resourceCCENodeAttachV3Create(d *schema.ResourceData, meta interface{}) err
...
+		}
+		_, err = stateConf.WaitForState()
+		if err != nil {
+			return fmtp.Errorf("Error resetting HuaweiCloud CCE Node: %s", err)
+		}
+
+		return resourceCCENodeV3Read(d, config)
+
+	} else {
+		return resourceCCENodeV3Update(d, config)
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_cce_node_attach.go:466]: how about **os** and **tags** are both changed?

File: huaweicloud/resource_huaweicloud_cce_node_attach.go (hunk 1 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_cce_node_attach.go]: fixed

File: huaweicloud/resource_huaweicloud_cce_node_attach.go (hunk 2 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_cce_node_attach.go:466]: if os is changed, the node will be reset, all parameters of the node will be set again

File: huaweicloud/resource_huaweicloud_cce_node_attach.go [hunk 3]
```diff
@@ -330,7 +389,86 @@ func resourceCCENodeAttachV3Create(d *schema.ResourceData, meta interface{}) err
...
+		resetOpts.NodeList = append(resetOpts.NodeList, resetNode)
+
+		logp.Printf("[DEBUG] Reset node Options: %#v", resetOpts)
+		// Add loginSpec here so it wouldn't go in the above log entry
+		var loginSpec nodes.LoginSpec
+		if hasFilledOpt(d, "key_pair") {
+			loginSpec = nodes.LoginSpec{
+				SshKey: d.Get("key_pair").(string),
+			}
+		} else if hasFilledOpt(d, "password") {
```
Comment (Lance52259) [inline huaweicloud/resource_huaweicloud_cce_node_attach.go]: declare `else` just ok

Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_cce_node_attach.go]: fixed

**Pull request merged**